lib/parse/parser_interpreter.es6.js

```diff
@@ -0,0 +1,299 @@
+          name: 'key',
+        },
+        {
+          typeName: 'String',
+          documentation: 'The string value in a "key:value" query fragment.',
+          name: 'value',
+        },
+      ],
+    },
+    {
+      name: 'interpretKeyword',
+      returns: 'foam.mlang.predicate.AbstractPredicate',
+      args: [
+        {
+          typeName: 'String',
+          documentation: 'The keyword in a keyword query fragment.',
+          name: 'kw',
+        },
+      ],
+    },
+  ],
+});
+
+foam.CLASS({
+  name: 'DefaultQueryInterpreter',
+  package: 'org.chromium.parse',
+  implements: [
+    'foam.mlang.Expressions',
+    'org.chromium.parse.QueryInterpreter',
+  ],
+
+  documentation: `Basic interpreter that matches key/values to properties and
+      interprets keywords in the natural way.`,
+
+  requires: ['foam.core.Property'],
+
+  methods: [
+    function interpretKeyValue(key, value, parser) {
+      const property = parser.of.getAxiomByName(key);
+      if (!property || !this.Property.isInstance(property)) {
+        return this.delegate ?
+            this.delegate.interpretKeyValue(key, value, parser) :
+            this.interpretKeyword(`${key}:${value}`, parser);
+      }
+      return this.EQ(property, value);
+    },
+    function interpretKeyword(kw, parser) {
+      return this.KEYWORD(kw);
+    },
+  ],
+});
+
+foam.CLASS({
+  name: 'ReleaseApiQueryInterpreter',
+  package: 'org.chromium.parse',
+  implements: [
+    'foam.mlang.Expressions',
+    'org.chromium.mlang.GridExpressions',
+    'org.chromium.mlang.ParseExpressions',
+    'org.chromium.parse.QueryInterpreter',
+  ],
+
+  documentation: `Interpreter for handling release key matching and
+      count-of-selected-releases. E.g.,
+
+      count:3 => "Exactly 3 selected releases support API"
+
+      fir59win10:0 => [prefix match alphabetical and numeric groups]
+                   => "Not shipped (last 0 is false) in Firefox 59 Windows 10"
+
+      chr60:t => [prefix match alphabetical and numeric groups]
+              => "Shipped (t for false) in Chrome 60 on all known platforms"`,
+
+  requires: [
+    'org.chromium.apis.web.CompatProperty',
+    'org.chromium.parse.DefaultQueryInterpreter',
+  ],
+  imports: [
+    'selectable',
+    'selected',
+  ],
+
+  properties: [
+    {
+      name: 'delegate',
+      factory: function() {
+        return this.DefaultQueryInterpreter.create();
+      },
+    },
+  ],
+
+  methods: [
+    function interpretKeyValue(key, value, parser) {
+      // When delegate is set and interpretation is null, pass to delegate for
+      // interpretation.
+      const ret = this.delegate ?
+          (v => v === null ?
+              this.delegate.interpretKeyValue(key, value, parser) : v) :
+          (v => v === null ?
+              this.interpretKeyword(`${key}:${value}`, parser) : v);
+
+      if (key === 'count') {
+        return ret(this.interpretCount(value, parser));
```

Comment:
You call the delegate twice here. Was that intentional?

Reply:
Not really, just defensive programming. We will only invoke the delegate twice if both `this` and `delegate` produce `null` (causing fallback in `interpretCount` and in `ret`). It's defensive because if `interpretCount` is changed to no longer fall back on the delegate, it'll still happen at least once in this code path.